### DISCUSSION TITLE
CVRP with supply constraint at depots

### DISCUSSION THREAD

**Discussion by lockeee** [google/or-tools#3170] [Routing (and legacy CP) questions]
Hello everybody,

I am trying to solve a version of the CVRP, where the supply of goods at the depot the vehicles are picking up and delivering is increasing over time. There is only one kind of good and it is "arriving" at depot at rate x (e.g 60) per hour, starting at 8:00 in the morning. I want to make sure every time a vehicle leaves the depot (or a fake extra depot node for reloading) the sum of goods that have already left the depot is smaller than the amount that has arrived at the depot by this time. 

Additionally the vehicles have different speeds and capacities and there are time window constraints.

I thought about adding e. g. 6 extra depots nodes per hour and assigning time windows (start_of_day+10_min*n, end_of_day) to simulate step by step supply of goods. Now i give the other nodes negative demand and the fake depots positive ones, to force the vehicles to first go to some depots to "load" and then unload at nodes. Obviously I allow that fake depots may be dropped.

Sadly the solver wont find any solutions anymore as soon as i go higher than 2 vehicles and 15 locations.

Could the same be implemented in another way? I've looked through git and stack overflow discussions and couldn't find any examples of my problem linked to or-tools, but I am sure I can't be the first to have asked this question.

I will supply code if needed, but i mostly adapted [cvrp_reload](https://github.com/google/or-tools/blob/master/ortools/constraint_solver/samples/cvrp_reload.py) to my needs.

Thanks in advance, and thank you in general for the amazing work you are doing here.
Lockeee

**Comment by Mizux** [comment]
When you say the solver didn't find a solution, did you means you didn't get any solution or the solution is not good enough with lots of deliveries still dropped ?
Did you enable the guided local search and the log search ?

note: I'll be happy to have a gist to reproduce it...

**Reply by lockeee** [reply]
It did not get any solution.
I did enable guided local search and i  was logging my results.

It's quite woven into existing software but i'll try to build a smallest non working example for you today.
Thanks for the quick reply.

**Reply by Mizux** [reply]
sometime the issue is the first solution strategy may have difficulties to find a first solution but allowing to drops nodes may help the solver to find a first (ugly) solution then improvide it during the local search...
